### DISCUSSION TITLE
One FormProvider, many forms

### DISCUSSION THREAD

**Discussion by sebaveg** [react-hook-form/react-hook-form#7206] [General] [answered]
Hello everyone

I have a StepWizard with some steps in a modal. In every step I have a form. I would like get context of forms in Modal for disable or enable next button.

So this is my code:
```js
<Modal {...modalProps}>
  <FormProvider {...methods}>
    <StepWizard>
      <FirstForm />
      <SecondForm />
      <ThirdForm />
      {...}
    </<StepWizard>
  </FormProvider>
</Modal>
```

So, FirstForm, SecondForm, ThirdForm have <form> every one.

In the modal I get the context:

```js
// modal.js
const { control } = useFormContext();
const { isDirty, isSubmitting } = useFormState({ control });
const disabledNextButton = !isDirty || isSubmitting;
return (
         <Button
              disabled={disabledNextButton}
          >
 )
```

This is correct? Or, Should I have one <FormProvider > in every form.

Thanks.

**Comment by bluebill1049** [comment, marked as answer]
Should be per form per context.
